### ISSUE TITLE
Too much information in events when pods fail to find a fit

### ISSUE THREAD

**Issue by derekwaynecarr** [kubernetes/kubernetes#27189]
If a pod fails to fit a node in the cluster, we create an event that explains why.  That is good.

My concern is that the event leaks potentially too much information or is just way too verbose for large clusters.  

The event that is produced is like the following:

```
Type=Warning
Reason=FailedScheduling
Message=pod (name) failed to fit in any node
  fit failure on node {node-a}: Insufficient CPU
  fit failure on node {node-b}: Out Of Disk
  fit failure on node {node-c}: Insufficient Memory
  fit failure on node {node-d}: Insufficient CPU
  fit failure on node {node-e}: Memory Pressure
  ... <repeats for each node in the cluster> ...
```

So if you have a large cluster size, but you create a pod that is larger than any node in that cluster, you get a really big event explaining that for each node.

I am worried that there are clusters where this may allow users to get too much information about nodes.  For example, clusters that secure access to create pods separately from viewing nodes.

Even then, it seems like we should collapse the message more:

failed to fit any node, X had InsufficientCPU, Y had Insufficient Memory, Z had ... so the message stays brief and the event can actually provide value to the user in a reasonable manner.

/cc @davidopp @bgrant0607 @smarterclayton @kubernetes/rh-cluster-infra @kubernetes/sig-scheduling

**Comment by smarterclayton** [comment]
How much output is generated by a 2k node cluster when it fails to find a
fit?  Also a security attack vector (define impossible conditions and then
schedule, which causes events to get created that are way too big)?

**Comment by derekwaynecarr** [comment]
@smarterclayton - i didnt try on a huge cluster, but it would be a line per node.  so....

**Comment by derekwaynecarr** [comment]
area we should think about changing:
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/generic_scheduler.go#L51

**Comment by davidopp** [comment]
ref/ #20064

**Comment by davidopp** [comment]
We should separate the security issue from the usability issue. The usability issue should be addressed by #20064 and is pretty easy. (It was supposed to be in 1.3 but didn't get it. We'll definitely have it in 1.4)

I'm not sure how to handle the other issue you mentioned. Some kind of flag that enables or disables how verbose the "why it won't schedule" information is, depending on the cluster security policy?

**Comment by HaiyangDING** [comment]
CC @HaiyangDING

I will keep this in mind when implementing #20064

**Comment by timothysc** [comment]
Fixed by: #35932
